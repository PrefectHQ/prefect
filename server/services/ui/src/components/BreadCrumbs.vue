<script>
export default {
  props: {
    crumbs: {
      type: Array,
      required: true
    }
  }
}
</script>

<template>
  <span>
    <span v-for="crumb in crumbs" :key="crumb.text">
      <RouterLink
        class="link"
        :to="crumb.route"
        :data-cy="'breadcrum-link|' + crumb.text"
      >
        {{ crumb.text }}
      </RouterLink>
      <v-icon small class="arrow-position">
        chevron_right
      </v-icon>
    </span>
  </span>
</template>

<style lang="scss" scoped>
.arrow-position {
  bottom: 1px;
  position: relative;
}
</style>
