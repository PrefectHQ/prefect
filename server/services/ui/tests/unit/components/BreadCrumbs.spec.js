import { shallowMount } from '@vue/test-utils'

import BreadCrumbs from '@/components/BreadCrumbs'

import Vue from 'vue'
Vue.config.silent = true

describe('BreadCrumbs', () => {
  test('matches snapshots', () => {
    const wrapper = shallowMount(BreadCrumbs, {
      propsData: {
        crumbs: [
          {
            route: {
              name: 'flow',
              params: { id: 'some-flow-id' }
            },
            text: 'second link'
          },
          {
            route: {
              name: 'flow-run',
              params: { id: 'some-flow-run-id' }
            },
            text: 'third-link'
          }
        ]
      }
    })

    expect(wrapper.element).toMatchSnapshot()
  })
})
