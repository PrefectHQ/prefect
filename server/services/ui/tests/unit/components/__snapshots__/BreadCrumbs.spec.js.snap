// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs matches snapshots 1`] = `
<span>
  <span>
    <routerlink
      class="link"
      data-cy="breadcrum-link|second link"
      to="[object Object]"
    >
      
      second link
    
    </routerlink>
     
    <v-icon
      class="arrow-position"
      small=""
    >
      
      chevron_right
    
    </v-icon>
  </span>
  <span>
    <routerlink
      class="link"
      data-cy="breadcrum-link|third-link"
      to="[object Object]"
    >
      
      third-link
    
    </routerlink>
     
    <v-icon
      class="arrow-position"
      small=""
    >
      
      chevron_right
    
    </v-icon>
  </span>
</span>
`;
